### DISCUSSION TITLE
Issue with local environment of Sunbird-ED portal

### DISCUSSION THREAD

**Discussion by 5Amogh** [Sunbird-Ed/Community#449] [General Discussion]
Hi @vinukumar-vs 

In the local environment of the Sunbird-ED portal some of the APIs fail with 403 forbidden error (with response message
: "You cannot consume this service" and You do not have permission to perform this operation ) and we have to hard reload the web application until we get the successful response and this is causing delay in the development as this is the case with every new build of the local at times takes 30-40 mins, those API URLs are mentioned below, Please check and help us, Thank you.

1. `learner/user/v1/consent/read`,
2. `/learner/user/v1/managed`,
3. `/learner/notification/v1/feed/read/:uid`
4.`/kendra/entities/mlcore/v1/entityTypesByLocationAndRole/:stateId`,
5. `kendra/user-extension/mlcore/v1/programsByPlatformRoles`
Cc: @kiranharidas187

**Comment by rajeshkumaravel** [comment]
@5Amogh
Can you please share more information regarding environment in which local setup is used and logged in User roles and which scenario / workflow these APIs are failing 
CC: @vinukumar-vs @rajeevsathish @Krishnaj20  @chitranshu-keshav

**Comment by 5Amogh** [comment]
@rajeshkumaravel
We are using staging environment in local and the logged-in user has the following roles: BOOK_CREATOR, BOOK_REVIEWER, CONTENT_CREATOR, CONTENT_REVIEWER, COURSE_ADMIN, COURSE_MENTOR, PROGRAM_DESIGNER, PROGRAM_MANAGER, REPORT_ADMIN, REPORT_VIEWER.
In the resources, profile, admin dashboard, program dashboard, observation, my reports pages these APIs are failing.
Attaching sample screen recordings for your reference please check.

[Screencast from 29-03-23 02:30:50 PM IST.webm](https://user-images.githubusercontent.com/93114621/228493004-84bf8688-bd6e-4242-8be6-b85e7cff5a5d.webm)

[Screencast from 29-03-23 02:35:07 PM IST.webm](https://user-images.githubusercontent.com/93114621/228493841-3a16de32-6b20-4fce-bc87-6f42a186dd9e.webm)

Cc: @kiranharidas187

**Reply by 5Amogh** [reply]
@rajeshkumaravel Did you get the time to check on this? Please let us know if anything is needed from our end.
Cc: @kiranharidas187

**Reply by rajeshkumaravel** [reply]
@5Amogh 
This is known issue, since local session is not valid in upstream and due to RBAC some of the APIs will not work in local setup.
CC: @swayangjit @vinukumar-vs @rajeevsathish @Krishnaj20 @chitranshu-keshav

**Reply by kiranharidas187** [reply]
@rajeshkumaravel The API works locally but we will have to keep reloading for a long while to make it work, hence wasting a lot of time.  Is there any way to fix this so that we can save a lot of time?

**Reply by rajeshkumaravel** [reply]
@kiranharidas187 
As of now there is no work around / fix available
CC: @swayangjit @vinukumar-vs @rajeevsathish @Krishnaj20 @chitranshu-keshav

**Reply by kiranharidas187** [reply]
@rajeshkumaravel  when can we expect it?

**Reply by vinukumar-vs** [reply]
@kiranharidas187 is it a major blocker? We have other priority items on the bucket now. Hence we have not prioritised this item at all, as it is not a major blocker for the dev team atleast.

Please let us know how frequently are you facing the issue of reload? 
We are planning to point all the developer testing to dev infra. As part of this, we will try to address after Angular migration(release-6.0.0).

If you have any solution please share it. If it is quick solution, team can help to resolve it.

**Reply by kiranharidas187** [reply]
@vinukumar-vs  Its a major issue. It will be helpful if this can be taken on priority after the migration. After each code change, we are now spending 5 mins reloading the page. On some days it will take up to 30 mins of reload.

We tried to identify the root cause but could not understand it.

**Reply by Krishnaj20** [reply]
@kiranharidas187 - Pls table this in next week's ED working group call. Lets see if other participants from the working group can contribute towards resolving this issue.

@Prateek-slokam @chitranshu-keshav @SanthoshVasabhaktula @parthlawate

**Comment by Krishnaj20** [comment]
@5Amogh - can you let us know which contributor you belong to?

**Reply by 5Amogh** [reply]
@Krishnaj20 
sure, i am from ShikshaLokam.
